tests/unit/amo/components/TestSearchForm.js

```diff
@@ -1,179 +1,386 @@
-    sinon.assert.notCalled(router.push);
-    input.value = 'searching';
-    Simulate.click(root.submitButton);
-    sinon.assert.calledWith(router.push, {
-      pathname: '/de/firefox/search/',
-      query: { q: 'searching' },
+    it('passes addonType when set', () => {
+      wrapper = mountComponent({
+        addonType: ADDON_TYPE_EXTENSION,
+      });
+
+      sinon.assert.notCalled(router.push);
+      wrapper.find('input').simulate('change', createFakeChangeEvent('& 26 %'));
+      wrapper.find('button').simulate('click');
+      sinon.assert.calledWith(router.push, {
+        pathname: '/de/firefox/search/',
+        query: { q: '& 26 %', type: ADDON_TYPE_EXTENSION },
+      });
     });
-  });
 
-  it('encodes the value of the search text', () => {
-    sinon.assert.notCalled(router.push);
-    input.value = '& 26 %';
-    Simulate.click(root.submitButton);
-    sinon.assert.calledWith(router.push, {
-      pathname: '/de/firefox/search/',
-      query: { q: '& 26 %' },
+    it('does not set type when it is not defined', () => {
+      sinon.assert.notCalled(router.push);
+      input.simulate('change', createFakeChangeEvent('searching'));
+      wrapper.find('button').simulate('click');
+      sinon.assert.calledWith(router.push, {
+        pathname: '/de/firefox/search/',
+        query: { q: 'searching' },
+      });
     });
-  });
-});
 
-describe('SearchForm mapStateToProps', () => {
-  it('passes the api through', () => {
-    const { store } = dispatchSignInActions();
+    it('encodes the value of the search text', () => {
+      sinon.assert.notCalled(router.push);
+      input.simulate('change', createFakeChangeEvent('& 26 %'));
+      wrapper.find('button').simulate('click');
+      sinon.assert.calledWith(router.push, {
+        pathname: '/de/firefox/search/',
+        query: { q: '& 26 %' },
+      });
+    });
 
-    const state = store.getState();
+    it('updates the state when props update', () => {
+      wrapper = mountComponent({ query: '' });
+      expect(wrapper.state('searchValue')).toEqual('');
 
-    expect(mapStateToProps(state).api).toEqual(state.api);
-  });
+      wrapper.setProps({ query: 'foo' });
+      expect(wrapper.state('searchValue')).toEqual('foo');
+    });
+
+    it('updates the state when user is typing', () => {
+      wrapper = mountComponent({ query: '' });
+      expect(wrapper.state('searchValue')).toEqual('');
+
+      wrapper.find('input').simulate('change', createFakeChangeEvent('foo'));
+      expect(wrapper.state('searchValue')).toEqual('foo');
+
+      wrapper.find('input').simulate('change', createFakeChangeEvent(undefined));
+      expect(wrapper.state('searchValue')).toEqual('');
+    });
+
+    it('fetches suggestions on focus', () => {
+      const dispatch = sinon.spy();
+      wrapper = mountComponent({
+        query: 'foo',
+        dispatch,
+      });
+      // no call to handleSuggestionsFetchRequested() until the input has focus,
```

Comment:
Try to use complete sentences in comments to make them easier to read. I think this would be "Expect no call to ..." ?